packages/open-truss/src/configuration/engine-v1.tsx

```diff
@@ -99,7 +102,11 @@ export function engineV1(
       }
 
       if (subFrame === undefined) {
-        return <Component key={i} {...props} />
+        if (data) {
```

Comment:
If I'm understanding this PR correctly, the changes make it so that if `data` is specified in the yaml the engine passes it to `/api/synchronous-uqi-query` to query results then renders the component with the data. Given the following yaml in your PR, it will send the trino query to `/api/synchronous-uqi-query` and render the component with the trino results returned.

```yaml
   - frame:
      data:
        query: select * from  tpch.sf1.customer limit 10
        source: trino-demo
      view:
        component: TrinoDemo
```

I think the engine / yaml annotation will likely need to be a little more flexible, but what you have here looks great to me to ship and we can make changes in a follow up PR 👍. 

A few thoughts about the `data` annotation in no particular order:
- I can see a world where pages have multiple frames that each have their own queries that could be run asynchronously. It would be great if we built in the ability to execute those queries asap so frames aren't serial / blocking
- I can also see a world where some frames may have dependencies on parent frames for data and we would need to support the ability to load them serially.
- As pages have many pieces of data, there likely will be many data variables. Making sure we handle namespacing and avoiding collisions will be an interesting problem to solve.
- These and many other wrinkles all don't have to be solved today and I'm sure we'll get to them eventually. 
- When we're a bit farther along it may be an interesting exercise to do a figjam brainstorming session to talk through more sophisticated aspects of the yaml rendering engine to help guide development.

Reply:
Agreed, things will get more complicated soon 😄